### PR TITLE
ROC-3821 bulk print fail alert

### PR DESCRIPTION
**Pull request by andrewwa-kainos**

### CODE REVIEW

File: bulk-print-fail-action-group.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module "cmc-bulk-print-fail-action-group" {
+  source = "git@github.com:hmcts/moj-module-action-group"
+  location = "${var.location}"
+  env = "${var.env}"
+  resourcegroup_name = "${azurerm_resource_group.rg.name}"
+  action_group_name = "Bulk Print Fail Alert - ${var.env}"
+  alert_short_name = "BPF_alert"
+  email_receiver_name = "Bulk Print Alerts"
+//  email_receiver_address = "cmc-bulk-print-alerts@hmcts.net" - disable for testing
```
Comment (timja) [inline bulk-print-fail-action-group.tf]: this should come from vault

File: bulk-print-fail-alert.tf [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module "cmc-bulk-print-fail-alert" {
+  source = "git@github.com:hmcts/moj-module-metric-alert"
+
+  app_insights_name = "${var.env == "prod" ? "cmc-prod" : "cmc-sandbox"}"
```
Comment (timja) [inline bulk-print-fail-alert.tf]: its just cmc-${env}

File: bulk-print-fail-alert.tf [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  alert_name = "Bulk print failure"
+  alert_desc = "Triggers when a bulk print failure event is received in a 5 minute poll."
+  app_insights_query = "customEvents | where name == \"Bulk print failed\""
+//  frequency_in_minutes = "5"
+//  time_window_in_minutes = "5"
+//  severity_level = "3"
+  action_group_name = "BPF_alert"
+//  trigger_threshold_operator = "GreaterThan"
+  trigger_threshold = "0"
+  resourcegroup_name = "${var.env == "prod" ? "cmc-prod" : "cmc-sandbox"}"
```
Comment (timja) [inline bulk-print-fail-alert.tf]: same as above

File: bulk-print-fail-alert.tf (hunk 2 above)
Comment (andrewwa-kainos) [reply bulk-print-fail-alert.tf]: I'd seen there are some environments without a corresponding application insights resource and somehow came to the conclusion that anything other than `prod` would work against `sandbox`... There's no `cmc-saat` application insights resource, for example. Confirm?

Comment (timja) [reply bulk-print-fail-alert.tf]: there's 
cmc-prod
cmc-demo
cmc-aat
cmc-sandbox
cmc-saat 
cmc-sprod

Some of which may have been deleted at some point
but it is cmc-${var.env}

Comment (andrewwa-kainos) [reply bulk-print-fail-alert.tf]: addressed

File: bulk-print-fail-alert.tf [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+  app_insights_name = "cmc-${var.env}"
+
+  alert_name = "Bulk print failure - CMC"
+  alert_desc = "Triggers when a bulk print failure event is received from CMC in a 5 minute poll."
+  app_insights_query = "customEvents | where name == \"Bulk print failed\""
+  frequency_in_minutes = 5
+  time_window_in_minutes = 5
+  severity_level = "3"
+  action_group_name = "Bulk Print Fail Alert - ${var.env}"
```
Comment (timja) [inline bulk-print-fail-alert.tf]: if you output this from `cnp-module-action-group` then you'll create a dependency which will enforce the order

File: bulk-print-fail-action-group.tf [hunk 5]
```diff
@@ -0,0 +1,11 @@
+module "cmc-bulk-print-fail-action-group" {
+  source = "git@github.com:hmcts/cnp-module-action-group"
+  location = "${var.location}"
+  env = "${var.env}"
+
+  resourcegroup_name = "${azurerm_resource_group.rg.name}"
+  action_group_name = "Bulk Print Fail Alert - ${var.env}"
+  short_name = "BPF_alert"
+  email_receiver_name = "Bulk Print Alerts"
```
Comment (singh-surjeet) [inline bulk-print-fail-action-group.tf:20]: Are short_name and email_receiver_name getting used? Custom email subject can be in this file. Values look bit scattered to me...

Comment (singh-surjeet) [reply bulk-print-fail-action-group.tf:20]: All values of this file can be inside be cmc-bulk-print-fail-alert.tf if needed. With unused values removed, all can be in one place I reckon.

Comment (andrewwa-kainos) [reply bulk-print-fail-action-group.tf:20]: Yeah they're all used. The custom email subject doesn't apply to the action group - it's a property of the alert that consumes the action group.

File: bulk-print-fail-alert.tf [hunk 6]
```diff
@@ -0,0 +1,18 @@
+module "cmc-bulk-print-fail-alert" {
+  source = "git@github.com:hmcts/cnp-module-metric-alert"
+  location = "${(var.env == "prod") ? var.location : "West Europe"}"
```
Comment (timja) [inline bulk-print-fail-alert.tf]: why is there a different location for prod?

Comment (andrewwa-kainos) [reply bulk-print-fail-alert.tf]: for `prod` I'm saying use whatever `location` is, but the default (`UK South`) isn't valid for an `env` of `sandbox`; and neither is the only location available for sandbox action groups, which is `global`. It was quite irritating.

Comment (timja) [reply bulk-print-fail-alert.tf]: it'll be the exact same for prod, app insights is available in west europe and not many other places

Comment (andrewwa-kainos) [reply bulk-print-fail-alert.tf]: So is there a change you want me to make? This complexity is purely to make it work in sandbox and test environments

Comment (timja) [reply bulk-print-fail-alert.tf]: its a change you have to make as it won't work as is in prod

File: bulk-print-fail-alert.tf [hunk 7]
```diff
@@ -0,0 +1,18 @@
+module "cmc-bulk-print-fail-alert" {
+  source = "git@github.com:hmcts/cnp-module-metric-alert"
+  location = "West Europe"
```
Comment (timja) [inline bulk-print-fail-alert.tf]: maybe use `appinsights_location` ?

**Pull request merged**